ivy/functional/backends/numpy/experimental/layers.py

```diff
@@ -1044,3 +1049,33 @@ def rfftn(
     if norm != "backward" and norm != "ortho" and norm != "forward":
         raise ivy.utils.exceptions.IvyError(f"Unrecognized normalization mode {norm}")
     return np.fft.rfftn(x, s, axes, norm).astype(np.complex128)
+
+
+def sliding_window(
+    input: np.ndarray,
+    kernel_size: Union[int, Sequence[int]],
+    /,
+    *,
+    stride: Union[int, Tuple[int, int]] = 1,
+    dilation: Union[int, Tuple[int, int]] = 1,
+    padding: Union[str, int, Tuple[int, int]] = "VALID",
+) -> np.ndarray:
+    k_size, stride, padding, dilation = map(
+        lambda x: tuple([x] * len(input.shape)) if isinstance(x, int) else x,
+        [kernel_size, stride, padding, dilation],
+    )
+
+    if isinstance(padding, str):
+        pads = _padtype_to_pads(input.shape, k_size, stride, padding)
+    else:
+        pads = padding
+
+    input = input.reshape((1, 1) + input.shape)
+    if dilation:
+        identity = ivy.array(0)
+        input = _dilate(input, dilation, identity)
+
+    view = _conv_view(input, [1, 1] + list(k_size), stride, pads, identity)[0]
+    view = ivy.reshape(view, (*view.shape[1 : 1 + len(k_size)], -1))
+
+    return view
```

Comment:
Currently you haven't added a paddle backend at all. You could move this compositional implementation to ivy and thus convert the function to a mixed function.
Also, since the jax backend is pretty much the same, we can omit that as well and have the function fall back to the main ivy composition. This will avoid code repetition.
So basically, we would end up with the torch and tensorflow backends, plus an ivy implementation that will run for the rest.

Reply:
Have found out a native function for paddle backend in https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/functional/unfold_en.html. Also, the latest commits include the changes as suggested.